smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatIntegrationTest.java

```diff
@@ -646,6 +648,102 @@ public void adminRevoked(EntityFullJid participant) {
         }
     }
 
+    /**
+     * Asserts that a user who gets kicked receives that change as a presence update
+     *
+     * <p>From XEP-0045 § 8.2:</p>
+     * <blockquote>
+     * The kick is performed based on the occupant's room nickname and is completed by setting the role of a
+     * participant or visitor to a value of "none".
+     *
+     * The service MUST remove the kicked occupant by sending a presence stanza of type "unavailable" to each kicked
+     * occupant, including status code 307 in the extended presence information, optionally along with the reason (if
+     * provided) and the roomnick or bare JID of the user who initiated the kick.
+     * </blockquote>
+     *
+     * @throws Exception when errors occur
+     */
+    @SmackIntegrationTest
+    public void mucPresenceTestForGettingKicked() throws Exception {
+        EntityBareJid mucAddress = getRandomRoom("smack-inttest");
+
+        MultiUserChat mucAsSeenByOne = mucManagerOne.getMultiUserChat(mucAddress);
+        MultiUserChat mucAsSeenByTwo = mucManagerTwo.getMultiUserChat(mucAddress);
+
+        createMUC(mucAsSeenByOne, "one-" + randomString);
+        final Resourcepart nicknameTwo = Resourcepart.from("two-" + randomString);
+        mucAsSeenByTwo.join(nicknameTwo);
+
+        final ResultSyncPoint<Presence, Exception> resultSyncPoint = new ResultSyncPoint<>();
+        mucAsSeenByTwo.addParticipantListener(kickPresence -> resultSyncPoint.signal(kickPresence));
+
+        mucAsSeenByOne.kickParticipant(nicknameTwo, "Nothing personal. Just a test.");
+        try {
+            Presence kickPresence = resultSyncPoint.waitForResult(timeout);
+            MUCUser mucUser = MUCUser.from(kickPresence);
+            assertNotNull(mucUser);
+            assertAll(
+                    () -> assertTrue(mucUser.getStatus().contains(MUCUser.Status.PRESENCE_TO_SELF_110), "Missing self-presence status code in kick presence"),
+                    () -> assertTrue(mucUser.getStatus().contains(MUCUser.Status.KICKED_307), "Missing kick status code in kick presence"),
+                    () -> assertEquals(MUCRole.none, mucUser.getItem().getRole(), "Role other than 'none' in kick presence")
+            );
+            if (mucUser.getItem().getJid() != null) {
```

Comment:
Save the result of `mucUser.getItem().getJid()` in a variable and use it below in the `assertEquals()`

Reply:
Fixed 👍